ArmRobot/Assets/Scripts/DomainRandomization.cs

```diff
@@ -0,0 +1,218 @@
+            colorRandomizer.ChangeColor();
+            listOfCylinders.Add(cylinder);
+        }
+        return listOfCylinders;
+    }
+
+    List<GameObject> CreateRandomListSpheres()
+    {
+        // this function is designed to create a list of a random number of spheres and instantiate them 
+        float minimumDistanceBetweenObjects = scaleObject + minimumDistanceWithoutObjects;
+        int randomNumberSpheres = 1 + Random.Range(1, nbMaxSpheres); 
+        List<GameObject> listOfSpheres = new List<GameObject>();
+        for (var i = 0; i < randomNumberSpheres; i++) {
+            GameObject sphere  = GameObject.CreatePrimitive(PrimitiveType.Sphere);
+
+            sphere.transform.position = new Vector3(robotMaxReach - 2*minimumDistanceWithoutObjects - minimumDistanceBetweenObjects, 
+                                                yAltitudeTable, robotMaxReach - 2*minimumDistanceWithoutObjects - minimumDistanceBetweenObjects*i);
+            
+            sphere.transform.localScale = new Vector3(scaleObject, scaleObject, scaleObject);
+            sphere.AddComponent<ColorRandomizer>();
+            sphere.AddComponent<RandomizerPositionObject>();
+            sphere.tag = "Sphere";
+            ColorRandomizer colorRandomizer = sphere.GetComponent<ColorRandomizer>();
+            colorRandomizer.ChangeColor();
+            listOfSpheres.Add(sphere);
+        }
+        return listOfSpheres;
+    }
+    List<GameObject> CreateRandomListLights()
+    {
+        // this function is designed to create a list of a random number of lights and instantiate them 
+        int randomNumberLights = 1 + Random.Range(1, nbMaxLights); 
+        List<GameObject> listOfLights = new List<GameObject>();
+        for (var i = 0; i < randomNumberLights; i++) {
+            GameObject light = new GameObject();
+            light.tag = "Light";
+
+            Light lightComp = light.AddComponent<Light>();
+
+            PointLightRandomization lightRandomization = GetComponent<PointLightRandomization>();
+            
+            light.transform.position = lightRandomization.PositionUpdate();
+            lightComp.intensity = lightRandomization.IntensityUpdate();
+            Vector4 color = lightRandomization.ColorUpdate();
+            lightComp.color = new Color(color[0], color[1], color[2], color[3]);
+            lightComp.range = 20f;
+
+            listOfLights.Add(light);
+        }
+        return listOfLights;
+    }
+
+    List<GameObject> ConcatenateListOfGameObjects(List<GameObject> list1, List<GameObject> list2){
+        // This function is designed to concatenate two lists of Gameobjects 
+        List<GameObject> listFinal  = new List<GameObject>();
+        foreach (GameObject gameobject in list1){
+            listFinal.Add(gameobject);
+        }
+        foreach (GameObject gameobject in list2){
+            listFinal.Add(gameobject);
+        }
+        return listFinal;
+    }
+    
+    void DestroyObjects(){
```

Comment:
Since creating game objects can be computationally expensive, I would recommend you create the max number of each game object, and simply enable and disable them instead. A disabled gameObject will not show up in the scene at runtime and its scripts will not run. See: https://docs.unity3d.com/ScriptReference/GameObject.SetActive.html